addons/stock/__init__.py

```diff
@@ -29,3 +29,16 @@ def _assign_default_mail_template_picking_id(cr, registry):
         company_ids_without_default_mail_template_id.write({
             'stock_mail_confirmation_template_id': default_mail_template_id.id,
         })
+
+def _uninstall_hook(cr, registry):
+    env = api.Environment(cr, SUPERUSER_ID, {})
+    try:
```

Comment:
```suggestion
    if not "purchase.order.line" in env.registry.models:
        return
    try:
```
avoid KeyError in case purchase is not installed 🤔